editors/code/src/installation/fetch_latest_artifact_metadata.ts

```diff
@@ -21,19 +21,35 @@ export async function fetchLatestArtifactMetadata(
 
     console.log("Issuing request for released artifacts metadata to", requestUrl);
 
-    const response: GithubRelease = await fetch(requestUrl, {
-            headers: { Accept: "application/vnd.github.v3+json" }
-        })
-        .then(res => res.json());
-
-    const artifact = response.assets.find(artifact => artifact.name === artifactFileName);
-
-    if (!artifact) return null;
+    try {
+        const ghReleaseResponse = await fetch(requestUrl, {
+                headers: { Accept: "application/vnd.github.v3+json" }
+            });
+
+        if (!ghReleaseResponse.ok) {
+            const responseText = await ghReleaseResponse.text();
+            console.log(`Could not fetch release metadata, cannot proceed`, {
+                response: responseText,
+                code: ghReleaseResponse.status,
+                status: ghReleaseResponse.statusText
+            });
+            return null;
+        }
+
+        const response: GithubRelease = await ghReleaseResponse.json();
```

Comment:
Its better to remove `try/catch` entirely and only leave this `if ` statement, all other code is not supposed to throw anyhow... Also, it would be more useful if we logged instead of `statusText` (which is just computed trivially from `status`) `headers` information (github passes rate-limit information in http headers)

Reply:
friendly ping @kwonoj